src/nifgen/examples/nifgen_script.py

```diff
@@ -3,13 +3,52 @@
 import argparse
 import nifgen
 import numpy as np
-from scipy import signal
+
+import math
 import sys
 import time
 
 number_of_points = 256
 
 
+def equivalent_phase(phase):
```

Comment:
Lines 14 - 87 is dedicated to creating waveforms. It uses numpy which is not a requirement for using `nifgen`.
That's excessive, the point of the examples is not to teach customers how to create waveforms but how to use `nifgen`.

The following code snippet creates all the waveforms using only built in modules `math` and `random`, each in one line. I _think_ they are correct :) at least they looked correct in Excel.
One difference is that the noise is no longer gaussian distribution, just uniform distribution. That's fine just need to rename things in the example.

```
import math
import random

number_of_samples = 50;

sine = [math.sin(math.pi * 2 * x / number_of_samples) for x in range(number_of_samples)]
square = [1.0 if x < number_of_samples/2 else -1.0 for x in range(number_of_samples)]
ramp_up = [2.0 * x / number_of_samples - 1.0 for x in range(number_of_samples)]
ramp_down = [-1.0 * x for x in ramp_up]
triangle = ramp_up[::2] + ramp_down[::2]
sawtooth = [(x / number_of_samples * 2) if x < number_of_samples/2 else (x / number_of_samples * 2) - 1 for x in range(number_of_samples)]
noise = [random.uniform(-1.0, 1.0) for x in range(number_of_samples)]
```

I much prefer the above as it's simpler and less verbose - not detracting from the point of the example.

Reply:
It seems a shame to throw away such a nice square function, but you have a point. I can certainly see how all this other code distracts from the actual example, especially with me adding to it. A single cycle (as opposed to an arbitrary number of cycles) for each waveform is probably also better for scripting.

A few things about the code you posted: 
* Ramps should start at 0
* Your triangle wave has more samples than the other wfms (which is okay) and has a -90 degree phase offset for no apparent reason
* The script doesn't actually call for both a triangle and sawtooth, but I don't mind adding it

Reply:
I have taken your feedback and removed the use of both scipy and numpy from the example.
I updated the wfm functions to be similar to what you have above, though not exactly the same in every case. I'm not going to add an actual triangle wfm. I'm just renaming the triangle wfm to sawtooth. As you more or less stated, it's not important to show users how to create every basic signal and I'd prefer all of our signals have 0 offset , 1 cycle (excluding the point at 2pi in case it's repeated) and be relatively simple.

I'm also sticking with 256 points, though I added a comment about it.

Reply:
What size do you recommend?

Reply:
I checked the specs for all currently supported devices. 512 samples should be enough to avoid underflow for anything except certain 5450/51 configurations. The 5450/51 is a major outlier with minimum waveform sizes above 2300 samples for some configurations.

It's also a major outlier in usage, since it's primarily an RF device, so I'm setting 512 as the waveform size. Let me know if you want me to increase it.

Reply:
The number 2096 popped in my mind.

Reply:
Yeah, I can set it to that. The 5450/51 has a burst trigger mode that has an even higher minimum waveform size, but I don't believe we're using that. https://www.ni.com/docs/en-US/bundle/pxie-5450-specs/page/specs.html

Really, it's hard for me to interpret, because I don't see anything for Script Trigger. Maybe script trigger would count as a single trigger? In which case the minimum size would be 2.

My one concern about increasing the size that much is that it might make the waveform noisy, when it's actually generated, due to a low delta.

Reply:
Increased to 2096 and marking this resolved.

Reply:
> My one concern about increasing the size that much is that it might make the waveform noisy, when it's actually generated, due to a low delta.

I don't follow.

Reply:
The signal is limited to the -1 to 1 range and as we increase the number of samples, the sine, sawtooth and ramps will decrease phase change between points, which decreases the amplitude change. So I worry about hardware precision limits causing this to make the signals appear noisy.

Perhaps it's an unfounded concern. I'm not an expert on how FGENs work.